src/lib/components/ChatMessage.svelte

```diff
@@ -132,6 +132,43 @@
         // Update the message
         message.bodyJson = JSON.stringify(plainContent);
 
+        console.log("Saving edited message:", message.bodyJson);
+        let messageJSON = JSON.parse(message.bodyJson) as JSONContent;
+
+        messageJSON.content = (messageJSON.content ?? []).map((block) => {
+          if (block.type === "paragraph" && Array.isArray(block.content)) {
+            let foundText = false;
+            block.content = block.content.filter((inline) => {
+              if (inline.type === "hardBreak" && !foundText) {
+                return false; // remove leading hardBreaks
+              }
+
+              if (inline.type === "text") {
+                inline.text = (inline?.text ?? "").replace(/^\s+/, ""); // remove leading spaces
+                if (inline.text.trim() === "") {
+                  return false; // remove text node if it's empty or all spaces
+                }
+                foundText = true;
+              }
+
+              return true;
+            });
+          }
+          return block;
+        });
+
+        console.log("messageJSON", messageJSON);
+
+        // If message is empty, don't save it
+        if (
+          messageJSON.content[0]?.content?.length === 0 ||
+          !messageJSON.content[0]?.content
+        ) {
+          toast.error("Message cannot be empty", { position: "bottom-end" });
```

Comment:
In this instance, I think it's clear to the user that they need to have a message before they can send it. Let's remove this toast.

```suggestion
          toast.error("Message cannot be empty", { position: "bottom-end" });
```